commit/plugin_roledon_e2e_test.go

```diff
@@ -182,6 +192,167 @@ func TestPlugin_RoleDonE2E_NoPrevOutcome(t *testing.T) {
+	chainsWithMsgs := []cciptypes.ChainSelector{s.sourceChains[0], s.sourceChains[1]}
+	t.Logf("Running test with Role DON Setup:\n%s", s)
+
+	allChainsConfig := make(map[cciptypes.ChainSelector]readerinternal.ChainConfig)
+	for _, ch := range append(s.sourceChains, s.destChain) {
+		allChainsConfig[ch] = readerinternal.ChainConfig{
+			FChain:         s.fChain[ch],
+			SupportedNodes: s.oracleIDsToPeerIDsSet(s.chainOracles[ch]),
+			Config:         chainconfig.ChainConfig{},
+		}
+	}
+
+	plugins := make([]ocr3types.ReportingPlugin[[]byte], 0, len(s.oracles))
+	for _, oracleID := range s.oracles {
+		deps := s.oracleDependencies[oracleID]
+
+		oracleChains := s.getChainsOfOracle(oracleID)
+		oracleSourceChainsSet := oracleChains.Clone()
+		oracleSourceChainsSet.Remove(s.destChain)
+		var oracleSourceChains []cciptypes.ChainSelector
+		if oracleSourceChainsSet.Cardinality() > 0 {
+			oracleSourceChains = oracleSourceChainsSet.ToSlice()
+			sort.Slice(oracleSourceChains, func(i, j int) bool { return oracleSourceChains[i] < oracleSourceChains[j] })
+		}
+
+		// Home Chain Expectations - Every oracle should be able to read
+		{
+			deps.homeChainReader.EXPECT().GetFChain().Return(s.fChain, nil)
+			deps.homeChainReader.EXPECT().GetAllChainConfigs().Return(allChainsConfig, nil)
+			deps.homeChainReader.EXPECT().GetSupportedChainsForPeer(mock.Anything).
+				RunAndReturn(func(id libocrtypes.PeerID) (mapset.Set[cciptypes.ChainSelector], error) {
+					supportedChainsOfOracle := s.getChainsOfOracle(s.peerIDToOracleID[id])
+					return supportedChainsOfOracle, nil
+				})
+			deps.homeChainReader.EXPECT().GetChainConfig(mock.Anything).
+				RunAndReturn(func(ch cciptypes.ChainSelector) (readerinternal.ChainConfig, error) {
+					return allChainsConfig[ch], nil
+				})
+		}
+
+		// Discovery and Sync - Out of scope of this test case
+		{
+			deps.ccipReader.EXPECT().DiscoverContracts(mock.Anything, mock.Anything).Return(nil, nil)
+			deps.ccipReader.EXPECT().Sync(mock.Anything, mock.Anything).Return(nil)
+		}
+
+		// Source Chain Expectations - Makes sure only oracles that support specific source chains are reading them.
+		{
+			if len(oracleSourceChains) > 0 && mapset.NewSet(oracleSourceChains...).ContainsAny(chainsWithMsgs...) {
+				deps.ccipReader.EXPECT().MsgsBetweenSeqNums(mock.Anything, mock.Anything, mock.Anything).
+					RunAndReturn(func(ctx context.Context, selector cciptypes.ChainSelector, numRange cciptypes.SeqNumRange) ([]cciptypes.Message, error) {
+						require.True(t, oracleChains.Contains(selector))
+						return []cciptypes.Message{}, nil // merkle roots consensus and computation is out of scope for this test
+					})
+			}
+		}
+
+		// Dest Chain Expectations - Makes sure only oracles that support the destination chain are reading it.
+		{
+			if oracleChains.Contains(s.destChain) {
+				deps.ccipReader.EXPECT().GetLatestPriceSeqNr(mock.Anything).Return(53, nil) // <-- still inflight (less than 54)
+			}
+		}
+
+		p := NewPlugin(
```

Comment:
nit, but can we extract this to something like `createPluginForTests(deps Deps, oracleId int)`. 
It pollutes the test case with unnecessary details